src/script/script.cpp

```diff
@@ -5,18 +5,11 @@
 
 #include "script.h"
 
+#include "script/interpreter.h"
```

Comment:
Oh, I hadn't noticed the circular dependence...maybe an additional moveonly commit would solve that?
mhmm, moving CheckSignatureEncoding and the flags from interpreter to script would be an option. Maybe move the flags to consensus/consensus instead?
Another option is not having anything depending on the flags in script/script: CScript methods can be turned into functions taking a CScript as parameter.
Or maybe we can just let the circular dependency be for now.
Mhmm, I would like to know what other people think about this...I still don't even know what option I like more.